cmd/podman/pod_kill.go

```diff
@@ -64,30 +60,9 @@ func podKillCmd(c *cli.Context) error {
 		killSignal = uint(sysSignal)
 	}
 
-	if c.Bool("all") {
-		pods, err = runtime.Pods()
-		if err != nil {
-			return errors.Wrapf(err, "unable to get pods")
-		}
-	}
-	if c.Bool("latest") {
-		pod, err := runtime.GetLatestPod()
-		if err != nil {
-			return errors.Wrapf(err, "unable to get latest pod")
-		}
-		pods = append(pods, pod)
-	}
-	for _, i := range args {
-		pod, err := runtime.LookupPod(i)
-		if err != nil {
-			logrus.Errorf("%q", lastError)
-			if lastError != nil {
-				logrus.Errorf("%q", lastError)
-			}
-			lastError = errors.Wrapf(err, "unable to find pods %s", i)
-			continue
-		}
-		pods = append(pods, pod)
+	pods, lastError := getPodsFromContext(c, runtime)
+	if len(pods) == 0 && lastError != nil {
```

Comment:
Do we need the length check here? @rhatdan I know you were in favor, but I think just checking for errors when retrieving containers is sufficient?

Reply:
yeah because the error being nil doesn't necessarily mean no pods were retrieved, only that some may have failed. If we return we can't kill those that succeeded.

Reply:
We shouldn't return an error here unless it's fatal - if it's not fatal, print it in getPodsFromContext() and return a nil error

Reply:
Shouldn't an error be returned if one of the pods wasn't found? There aren't really any "fatal" errors, but the error code should reflect something failed if it did. I can't think of an error that shouldn't be returned to the commands that would come from getPods...

Reply:
Looking into this more, if we don't propogate the error out, we'll return 0 on failure even though some pods were not found, which is not good...

It looks like this way is the best way of doing business, even if it's a bit gross to return an error and then seemingly ignore it.

Sorry for the noise...

Reply:
Personally I feel the best way to go about it is to make a comment and say something like an error being returned isn't fatal unless there are no pods, in which case the loop will be skipped. I'm fine with it as is though

Reply:
Comments for clarity are always welcome!